spring-integration-core/src/main/java/org/springframework/integration/channel/PartitionedChannel.java

```diff
@@ -65,7 +66,8 @@ public class PartitionedChannel extends AbstractExecutorChannel {
 	 * sent to this channel.
 	 */
 	public PartitionedChannel(int partitionCount) {
-		this(partitionCount, (message) -> message.getHeaders().get(IntegrationMessageHeaderAccessor.CORRELATION_ID));
+		this(partitionCount, (message) -> Objects.requireNonNull(
+				message.getHeaders().get(IntegrationMessageHeaderAccessor.CORRELATION_ID)));
```

Comment:
This is just nit-pick, but for me the code is more readable when we put everything after an arrow on a new line and have that as a single line.
I mean no breaks on the `requireNonNull(`